rotkehlchen/accounting/rules.py

```diff
@@ -0,0 +1,170 @@
+        taxable=False,  # this needs to become configurable as per https://github.com/rotki/rotki/issues/4341 since it used to respect the airdrop ledger action settings  # noqa: E501
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    result[get_accounting_rule_identifier(HistoryEventType.RECEIVE, HistoryEventSubType.REWARD)] = BaseEventSettings(  # noqa: E501
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    result[get_accounting_rule_identifier(HistoryEventType.STAKING, HistoryEventSubType.REWARD)] = BaseEventSettings(  # noqa: E501
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    result[get_accounting_rule_identifier(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)] = BaseEventSettings(  # noqa: E501
+        taxable=False,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='spend',
+    )
+    result[get_accounting_rule_identifier(HistoryEventType.STAKING, HistoryEventSubType.REMOVE_ASSET)] = BaseEventSettings(  # noqa: E501
+        taxable=False,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+class AccountingRulesManager:
+
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_aggregators: 'EVMAccountingAggregators',
+            pot: 'AccountingPot',
+    ):
+        self.database = database
+        self.aggregators = evm_aggregators
+        self.pot = pot
+        self.event_settings: dict[str, BaseEventSettings] = {}
+
+    def _query_db_rules(self) -> None:
+        """Query the accounting rules in the db and update event_settings with them"""
+        with self.database.conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT type, subtype, counterparty, taxable, count_entire_amount_spend, '
+                'count_cost_basis_pnl, method, accounting_treatment FROM accounting_rules',
+            )
+            for entry in cursor:
+                rule = BaseEventSettings.deserialize_from_db(entry[3:])
+                key = get_accounting_rule_identifier(
+                    event_type=HistoryEventType.deserialize(entry[0]),
+                    event_subtype=HistoryEventSubType.deserialize(entry[1]),
+                    counterparty=entry[2] if entry[2] != NO_ACCOUNTING_COUNTERPARTY else None,
```

Comment:
Question.

Is this the only place where the logic `entry[2] if entry[2] != NO_ACCOUNTING_COUNTERPARTY else None` is to be used?

If so then it's good to leave it as is. And by checking quickly with Ctrl-f it seems to be.

IF not, then we would need to move it in `get_accounting_rule_identifier` itself.